### ISSUE TITLE
[Bug][Android] Keyboard issue with LayoutNoLimits on Samsung devices with Android 11

### ISSUE THREAD

**Issue by rickclephas** [xamarin/Xamarin.Forms#14067]
### Description

On Samsung devices running Android 11 there is an issue with the `WindowSoftInputMode` when `WindowManagerFlags.LayoutNoLimits` is set. After closing the keyboard the app doesn't return to it's normal state and shows a black bar at the bottom of the screen:

### Steps to Reproduce

1. Create a new Forms project
2. Set the `WindowManagerFlags.LayoutNoLimits` in the `OnCreate` of the `MainActivity`:
```c#
Window?.AddFlags(WindowManagerFlags.LayoutNoLimits);
```
3. Add an `Entry` to the bottom of the `MainPage`:
```xaml
<ScrollView BackgroundColor="Purple">
    <StackLayout BackgroundColor="Green">
        <BoxView HeightRequest="1000"
                 Color="Red"/>
        <Entry BackgroundColor="Blue"/>
        <BoxView HeightRequest="100"
                 Color="Brown"/>
    </StackLayout>
</ScrollView>
```
4. Start the app and focus on the `Entry`
5. Now hide the keyboard

### Expected Behavior

The app is moved up when the keyboard is shown and moved back down when it is hidden.
**NOTE:** this is exactly how it works on e.g. a Google Pixel 3a XL running Android 11.

### Actual Behavior

There is a black bar at the bottom of the screen after the keyboard is hidden.

### Basic Information

- Version with issue: 5.0.0.2012
- Platform Target Frameworks: 
  - Android 11
- Affected Devices:
  - Samsung devices running Android 11

### Screenshots

![Screenshot_20210326-112448](https://user-images.githubusercontent.com/7353419/112618207-2eccb080-8e26-11eb-986d-9431afdd5717.jpg)

### Reproduction Link

[XFSamsungA11KeyboardBug.zip](https://github.com/xamarin/Xamarin.Forms/files/6211104/XFSamsungA11KeyboardBug.zip)

**Comment by rachelkang** [comment]
Wasn't able to repro on a Samsung device with Android 7 - will try to look for a Samsung device on Android 11 and try again

**Comment by rickclephas** [comment]
@rachelkang that's correct. I couldn't reproduce this on older Android versions either. So far it seems to only affect Samsung devices on Android 11.

I should also note that there is a difference in behavior between Android 10 and 11. On Android 10 the entry stays hidden behind the keyboard while on Android 11 it stays above the keyboard.

Maybe the new Android 11 APIs for IME transitions have something to do with this?

The behavior difference doesn't seem like a big problem to me but the bug on Samsung devices definitely is!

The only way to get rid of the black bar is to either focus on an entry that is always shown above the keyboard or switch between apps.

P.S. the behavior difference is the same with a native Android app but I wasn't able to reproduce the bug in a native app so that seems to be something Xamarin Forms related.
